rwengine/src/script/ScriptMachine.cpp

```diff
@@ -148,10 +148,15 @@ void ScriptMachine::executeThread(SCMThread &t, int msPassed)
 #if RW_SCRIPT_DEBUG
 		if (strcmp(t.name, "EIGHT") == 0)
 		{
-			printf("% 8s  %04x %04x % 25s", t.name, t.programCounter, opcode, code.signature.c_str());
+			printf("% 8s %01x %04x %04x % 25s", t.name, t.conditionResult, t.programCounter, opcode, code.signature.c_str());
 			for (auto& a : sca.getParameters())
 			{
-				printf(" %08x", a.integerValue());
+				if (a.type == SCMType::TString) {
+					printf(" %01x:% 8s", a.type, a.string);
+				}
+				else {
+					printf(" %01x:%08x", a.type, a.integerValue());
```

Comment:
This will often print out long error messages because the type is not handled properly.
We also need something for realValue();

Would it be possible to create something for this PR already?

I had a similar commit in my cleanup but I'd drop it in favor of this one.